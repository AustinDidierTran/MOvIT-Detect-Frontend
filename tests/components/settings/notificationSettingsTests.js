/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import MockAdapter from 'axios-mock-adapter';
import PropTypes from 'prop-types';
import React from 'react';
import axios from 'axios';
import configureMockStore from 'redux-mock-store';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import NotificationSettings from '../../../src/components/settings/notificationSettings';

Enzyme.configure({ adapter: new Adapter() });

function initializeMockAdapter() {
  const mock = new MockAdapter(axios);

  mock.onPost(`${URL}notificationSettings`).reply(200);
}

describe('NotificationTests Tests', () => {
  let wrapper;
  let store;
  const snoozeTimeSpy = sinon.spy();
  const isLedBlinkingEnabledSpy = sinon.spy();
  const isVibrationEnabledSpy = sinon.spy();
  const initialState = { applicationReducer: { language: 'en' } };
  const mockStore = configureMockStore();
  const props = {
    snoozeTime: 15,
    isLedBlinkingEnabled: true,
    isVibrationEnabled: true,
    changeSnoozeTime: (value) => { snoozeTimeSpy(value); },
    changeIsLedBlinkingEnabled: (value) => { isLedBlinkingEnabledSpy(value); },
    changeIsVibrationEnabled: (value) => { isVibrationEnabledSpy(value); },
    hasErrors: false,
  };

  initializeMockAdapter();

  beforeEach(() => {
    snoozeTimeSpy.resetHistory();
    isLedBlinkingEnabledSpy.resetHistory();
    isVibrationEnabledSpy.resetHistory();

    store = mockStore(initialState);
    wrapper = shallow(<NotificationSettings store={store} {...props} />).dive();
  });

  it('should have proptypes', () => {
    const actualValue = NotificationSettings.WrappedComponent.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
      header: PropTypes.object,
      snoozeTime: PropTypes.number.isRequired,
      isLedBlinkingEnabled: PropTypes.bool.isRequired,
      isVibrationEnabled: PropTypes.bool.isRequired,
      changeSnoozeTime: PropTypes.func.isRequired,
      changeIsLedBlinkingEnabled: PropTypes.func.isRequired,
      changeIsVibrationEnabled: PropTypes.func.isRequired,
      hasErrors: PropTypes.bool.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should trigger changeIsLedBlinkingEnabled when simulating a change on the led checkbox', () => {
    wrapper.setProps({ isLedBlinkingEnabled: false });

    wrapper.instance().enableLedBlinking();

    expect(isLedBlinkingEnabledSpy.calledOnce).toEqual(true);
    expect(isLedBlinkingEnabledSpy.getCalls()[0].args[0]).toEqual(true);
  });

  it('should triggger changeIsVibrationEnabled when simulating a change on the vibration checkbox', () => {
    wrapper.setProps({ isVibrationEnabled: false });

    wrapper.instance().enableVibration();

    expect(isVibrationEnabledSpy.calledOnce).toEqual(true);
    expect(isVibrationEnabledSpy.getCalls()[0].args[0]).toEqual(true);
  });

  it('should trigger changeSnoozeTime when changing the input value', () => {
    wrapper.instance().changeSnoozeTime(10);

    expect(snoozeTimeSpy.calledOnce).toEqual(true);
    expect(snoozeTimeSpy.getCalls()[0].args[0]).toEqual(10);
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
