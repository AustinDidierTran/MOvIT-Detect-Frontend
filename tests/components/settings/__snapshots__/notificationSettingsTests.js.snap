// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationTests Tests should match the snapshot 1`] = `
<div>
  <div>
    <Checkbox
      checked={true}
      className={null}
      disabled={false}
      id="enableLedBlinking"
      inputId={null}
      name={null}
      onChange={[Function]}
      onContextMenu={null}
      onMouseDown={null}
      readOnly={false}
      style={null}
      value={null}
    />
    <label
      htmlFor="enableLedBlinking"
    >
      settings.notification.enableLedBlinking.en
    </label>
  </div>
  <div>
    <Checkbox
      checked={true}
      className={null}
      disabled={false}
      id="enableVibration"
      inputId={null}
      name={null}
      onChange={[Function]}
      onContextMenu={null}
      onMouseDown={null}
      readOnly={false}
      style={null}
      value={null}
    />
    <label
      htmlFor="enableVibration"
    >
      settings.notification.enableVibration.en
    </label>
  </div>
  <div>
    <span>
      <i
        className="fa fa-info-circle"
        id="snoozeTimeToolTip"
      />
       
      settings.notification.snoozeTime.en
      :  
    </span>
    <Tooltip
      appendTo="body"
      escape={true}
      for="#snoozeTimeToolTip"
      hideDelay={null}
      onBeforeShow={null}
      positionStyle={null}
      showDelay={null}
      title="settings.notification.snoozeTimeToolTip.en"
      tooltipDisabled={false}
      tooltipEvent="hover"
      tooltipPosition="right"
      tooltipStyleClass={null}
    />
    <Spinner
      className={null}
      decimalSeparator="."
      disabled={false}
      id="value"
      max={60}
      maxlength={2}
      min={0}
      onBlur={[Function]}
      onChange={[Function]}
      readonly={false}
      size="3"
      step={1}
      style={null}
      thousandSeparator=","
      type="number"
      value={null}
    />
    <span>
        
      time.minutes.en
    </span>
  </div>
</div>
`;
